### PR TITLE
Change move to copy build dir during an upgrade

### PR DESCRIPTION
**Pull request by rafzei**

### CODE REVIEW

File: core/src/epicli/cli/engine/UpgradeEngine.py [hunk 1]
```diff
@@ -42,11 +42,8 @@ def backup_build(self):
         backup_dir_name = f'backup_{int(round(time.time() * 1000))}'
         backup_temp_dir = os.path.join(self.build_dir, '../', backup_dir_name)
         self.backup_build_dir = os.path.join(self.build_dir, backup_dir_name )
-        self.logger.info(f'Backing up old build files to "{self.backup_build_dir}"')
-        shutil.move(self.build_dir, backup_temp_dir, copy_function = shutil.copytree) 
-        if not os.path.exists(self.build_dir):
-            os.mkdir(self.build_dir)        
-        shutil.move(backup_temp_dir, self.backup_build_dir, copy_function = shutil.copytree)
+        self.logger.info(f'Backing up build dir to "{self.backup_build_dir}"')
+        shutil.copytree(self.build_dir, self.backup_build_dir)
```
Comment (seriva) [inline core/src/epicli/cli/engine/UpgradeEngine.py:46]: Did you consider the old 0.2.3 backup folders? Is this compatible with this?

File: core/src/epicli/cli/engine/UpgradeEngine.py [hunk 2]
```diff
@@ -42,11 +42,8 @@ def backup_build(self):
         backup_dir_name = f'backup_{int(round(time.time() * 1000))}'
         backup_temp_dir = os.path.join(self.build_dir, '../', backup_dir_name)
```
Comment (seriva) [inline core/src/epicli/cli/engine/UpgradeEngine.py:43]: Is backup_temp_dir  still used?

File: core/src/epicli/cli/engine/UpgradeEngine.py (hunk 1 above)
Comment (rafzei) [reply core/src/epicli/cli/engine/UpgradeEngine.py:45]: We are not supporting 0.2 versions anymore.

File: core/src/epicli/cli/engine/UpgradeEngine.py [hunk 3]
```diff
@@ -40,13 +40,9 @@ def backup_build(self):
...
         backup_dir_name = f'backup_{int(round(time.time() * 1000))}'
-        backup_temp_dir = os.path.join(self.build_dir, '../', backup_dir_name)
         self.backup_build_dir = os.path.join(self.build_dir, backup_dir_name )
-        self.logger.info(f'Backing up old build files to "{self.backup_build_dir}"')
-        shutil.move(self.build_dir, backup_temp_dir, copy_function = shutil.copytree) 
-        if not os.path.exists(self.build_dir):
-            os.mkdir(self.build_dir)        
-        shutil.move(backup_temp_dir, self.backup_build_dir, copy_function = shutil.copytree)
+        self.logger.info(f'Backing up build dir to "{self.backup_build_dir}"')
+        shutil.copytree(self.build_dir, self.backup_build_dir)
```
Comment (sk4zuzu) [inline core/src/epicli/cli/engine/UpgradeEngine.py:45]: Please take a look at this part of Python 3.7 documentation (https://docs.python.org/3.7/library/shutil.html#shutil.copytree):
```
shutil.copytree(src, dst, symlinks=False, ignore=None, copy_function=copy2, ignore_dangling_symlinks=False)
Recursively copy an entire directory tree rooted at src, returning the destination directory. The destination directory, named by dst, must not already exist;
```
It clearly states that the `directory must not exist`. Of course the name is a timestamp basically, but personally I would be extra paranoid here (but that just maybe me idk, please ignore if you think it's not a valid suggestion) \\:D/.

**Pull request merged**